compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/scene/ComposeSceneMediator.uikit.kt

```diff
@@ -169,65 +151,23 @@ private class SemanticsOwnerListenerImpl(
     }
 }
 
-private class RenderingUIViewDelegateImpl(
-    private val scene: ComposeScene,
-    private val sceneOffset: () -> Offset,
-) : SkikoRenderDelegate {
-    override fun onRender(canvas: Canvas, width: Int, height: Int, nanoTime: Long) {
-        canvas.withSceneOffset {
-            scene.render(asComposeCanvas(), nanoTime)
-        }
-    }
-
-    private inline fun Canvas.withSceneOffset(block: Canvas.() -> Unit) {
-        val sceneOffset = sceneOffset()
-        save()
-        translate(sceneOffset.x, sceneOffset.y)
-        block()
-        restore()
-    }
-}
-
-private class ComposeSceneMediatorRootUIView : UIView(CGRectZero.readValue()) {
-    override fun hitTest(point: CValue<CGPoint>, withEvent: UIEvent?): UIView? {
-        // forwards touches forward to the children, is never a target for a touch
-        val result = super.hitTest(point, withEvent)
-
-        return if (result == this) {
-            null
-        } else {
-            result
-        }
-    }
-}
-
-internal class ComposeSceneMediator(
-    private val container: UIView,
+internal abstract class ComposeSceneMediator(
```

Comment:
This part is not supposed to be abstract

Reply:
Then we have a god object.

Reply:
Then, it seems you're trying to do things that are not supposed to be done there